Classes/ios/Scanners/MTBBarcodeScanner.m

```diff
@@ -661,24 +673,30 @@ - (AVCaptureTorchMode)avTorchModeForMTBTorchMode:(MTBTorchMode)torchMode {
 
 - (void)freezeCapture {
     self.capturePreviewLayer.connection.enabled = NO;
-    
-    if (self.hasExistingSession) {
+
+    dispatch_async(self.privateSessionQueue, ^{
         [self.session stopRunning];
-    }
+    });
 }
 
 - (void)unfreezeCapture {
+    if (!self.session) {
+        return;
+    }
+
     self.capturePreviewLayer.connection.enabled = YES;
```

Comment:
I believe it is ok as the session and the layer are created and destroyed at the same times, and the preview layer will be (visibly) useless without a session.